nearcore/src/runtime/mod.rs

```diff
@@ -1542,6 +1580,73 @@ impl RuntimeAdapter for NightshadeRuntime {
         }
     }
 
+    fn build_state_for_split_shards(
+        &self,
+        shard_uid: ShardUId,
+        state_root: &StateRoot,
+        next_epoch_shard_layout: &ShardLayout,
+    ) -> Result<HashMap<ShardUId, StateRoot>, Error> {
+        let trie = self.get_tries().get_view_trie_for_shard(shard_uid);
+        let shard_id = shard_uid.shard_id();
+        let mut state_roots = {
+            let split_shards = next_epoch_shard_layout
+                .get_split_shards(shard_id)
+                .ok_or(ErrorKind::InvalidShardId(shard_id))?;
+            split_shards
+                .iter()
+                .map(|id| {
+                    (
+                        ShardUId::from_shard_id_and_layout(*id, &next_epoch_shard_layout),
+                        StateRoot::default(),
+                    )
+                })
+                .collect()
+        };
+
+        let state_root_node = trie.retrieve_root_node(state_root).map_err(|err| err.to_string())?;
```

Comment:
Why do we map err to string here?

Reply:
hmm I think I saw somewhere in the code that `StorageError` is converted to a string and I just followed that. I implemented the trait `From<StorageError>` for `Error` so now `StorageError` is converted to `Error` correctly